tools/config/__init__.py

```diff
@@ -1058,9 +1063,62 @@ def validate_config(self):
 
         Arguments: None
         """
+
         params, _ = self.get_config_data()
+        err_msg = ""
+
+        for name, param in sorted(params.items()):
+            min      = param.value_min
+            max      = param.value_max
+            accepted = param.accepted_values
+            value    = param.value
+
+            # Config parameters that are only defined but do not have a default
+            # value should not be range limited
+            if value is not None:
+                if (min is not None or max is not None) and (accepted is not None):
+                    err_msg += "\n%s has both a range and list of accepted values specified. Please only "\
+                                "specify either value_min and/or value_max, or accepted_values"\
+                                    % param
+                else:
+                    if re.match(r'^(0[xX])[A-Fa-f0-9]+$|^[0-9]+$', str(value)):
+                        # Value is a hexadecimal or numerical string value
+                        # Convert to a python integer and range check/compare to
+                        # accepted list accordingly
+
+                        if min is not None or max is not None:
+                            # Numerical range check
+                            # Convert hex strings to integers for range checks
+
+                            value = int(str(value), 0)
+                            min = int(str(min), 0) if min is not None else None
+                            max = int(str(max), 0) if max is not None else None
+
+                            if (value < min or (value > max if max is not None else False)):
+                                err_msg += "\nInvalid config range for %s, is not in the required range: [%s:%s]"\
+                                               % (param,
+                                                  min if min is not None else "-inf",
+                                                  max if max is not None else "inf")
+
+                        # Numerical accepted value check
+                        elif accepted is not None and value not in accepted:
+                           err_msg += "\nInvalid config range for %s, is not an accepted value: %s"\
```

Comment:
> Invalid config range for ...

Since this is checking the accepted list instead of the range, would this be a more appropriate error?

```suggestion
                           err_msg += "\nInvalid value for %s, is not an accepted value: %s"\
```